mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -27,15 +27,35 @@ private[spark] object BLAS extends Serializable {
 
   @transient private var _f2jBLAS: NetlibBLAS = _
   @transient private var _nativeBLAS: NetlibBLAS = _
+  @transient private val vectorSizeThreshold: Int = 256
 
-  // For level-1 routines, we use Java implementation.
+  // For level-3 routines, we use the native BLAS.
+  // if native BLAS is not properly configured in system, will automatically fallback to f2jBLAS
+  private[ml] def nativeBLAS: NetlibBLAS = {
```

Comment:
I don't think you need to move this or expand its visibility?

Reply:
I add a note in ml-guide to highlight nativeBLAS usage depends on proper configuration in system otherwise will fallback to java implementation.

Reply:
This still can move back to where it was